src/path.rs

```diff
@@ -230,6 +231,32 @@ impl ClioPath {
         }
     }
 
+    /// Adds an extension to the end of the [`self.file_name`](Path::file_name).
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use clio::ClioPath;
+    ///
+    /// let mut p = ClioPath::new("/tmp/log.txt")?;
+    ///
+    /// p.add_extension("gz");
+    /// assert_eq!(ClioPath::new("/tmp/log.txt.gz")?, p);
+    /// # Ok::<(), clio::Error>(())
+    /// ```
+    #[rustversion::before(1.70)]
+    pub fn add_extension<S: AsRef<OsStr>>(&mut self, extension: S) -> bool {
```

Comment:
If this moves forward, I would recommend having a single public function with a single piece of documentation that then calls into a shared function.  That shred function shouldn't accept any generic parameters (in general, any function that accepts generic parameters like `AsRef` should do this)